pkg/reconciler/common/common.go

```diff
@@ -43,7 +43,8 @@ func PipelineReady(informer informer.TektonPipelineInformer) (*v1alpha1.TektonPi
 		}
 		return nil, err
 	}
-	if ppln.GetStatus() != nil && strings.Contains(ppln.GetStatus().GetCondition(apis.ConditionReady).Message, v1alpha1.UpgradePending) {
+	readyCondition := ppln.GetStatus().GetCondition(apis.ConditionReady)
+	if readyCondition != nil && strings.Contains(readyCondition.Message, v1alpha1.UpgradePending) {
```

Comment:
In my environment, sometimes the Pipeline has not yet been reconciled, resulting in the status being empty. At this point, entering here may trigger a panic.

Reply:
@l-qing we need to keep the check ppln.GetStatus() != nil, otherwise it will panic on ppln.GetStatus().GetCondition, and refactor as you suggest ?

Reply:
Although it's not possible for nil here, it is indeed necessary to optimize it to reduce cognitive load.

https://github.com/tektoncd/operator/blob/f547f0924df772c81e7960959615deec57e1798d/pkg/apis/operator/v1alpha1/tektonpipeline_types.go#L30-L44